numpy/core/getlimits.py

```diff
@@ -471,6 +471,14 @@ class finfo:
     _finfo_cache = {}
 
     def __new__(cls, dtype):
+        if dtype is None:
+            warnings.warn(
+                "dtype cannot be None. This behavior "
+                "will raise an error in the future.",
```

Comment:
Thanks, the warning is indeed the correct path.  Could you add a test as well?  It can live in `test_deprecations.py`.

Please add `(Deprecated in NumPy 1.25)` or something similar to the message.  Could also lead with `finfo() dtype ...` just to make it even easier to find where it is coming from.

The last thing is that both here and in the test, we try to add a comment:
```
# Deprecated in NumPy 1.25, 2022-01-16
```
Or similar.  This makes it much easier to find and to act on in the future.